flutter/lib/src/frames_tracking/sentry_delayed_frames_tracker.dart

```diff
@@ -0,0 +1,303 @@
+        // finished processing
+        pause();
+      }
+    }
+    _resetCurrentFrame();
+  }
+
+  void _resetCurrentFrame() {
+    _currentFrameStartTimestamp = null;
+  }
+
+  /// Resumes the collecting of frames.
+  void resume() {
+    _isTrackingActive = true;
+  }
+
+  /// Pauses the collecting of frames.
+  void pause() {
+    _isTrackingActive = false;
+    _resetCurrentFrame();
+  }
+
+  /// Retrieves the frames the intersect with the provided [startTimestamp] and [endTimestamp].
+  @visibleForTesting
+  List<SentryFrameTiming> getFramesIntersecting(
+      {required DateTime startTimestamp, required DateTime endTimestamp}) {
+    return _delayedFrames.where((frame) {
+      // Fully contained or exactly matching
+      final fullyContainedOrMatching =
+          frame.startTimestamp.compareTo(startTimestamp) >= 0 &&
+              frame.endTimestamp.compareTo(endTimestamp) <= 0;
+
+      // Partially contained, starts before range, ends within range
+      final startsBeforeEndsWithin =
+          frame.startTimestamp.isBefore(startTimestamp) &&
+              frame.endTimestamp.isAfter(startTimestamp) &&
+              frame.endTimestamp.isBefore(endTimestamp);
+
+      // Partially contained, starts within range, ends after range
+      final startsWithinEndsAfter =
+          frame.startTimestamp.isAfter(startTimestamp) &&
+              frame.startTimestamp.isBefore(endTimestamp) &&
+              frame.endTimestamp.isAfter(endTimestamp);
+
+      return fullyContainedOrMatching ||
+          startsBeforeEndsWithin ||
+          startsWithinEndsAfter;
+    }).toList(growable: false);
+  }
+
+  /// Calculates the frame metrics based on start, end timestamps and the
+  /// delayed frames metrics. If the delayed frames array is empty then
+  /// only the total frames will be calculated.
+  SpanFrameMetrics? getFrameMetrics(
+      {required DateTime spanStartTimestamp,
+      required DateTime spanEndTimestamp}) {
+    final relevantFrames = getFramesIntersecting(
+        startTimestamp: spanStartTimestamp, endTimestamp: spanEndTimestamp);
+    final spanDuration =
+        spanEndTimestamp.difference(spanStartTimestamp).inMilliseconds;
+
+    // No slow or frozen frames detected
+    if (relevantFrames.isEmpty) {
+      return SpanFrameMetrics(
+          totalFrameCount:
+              (spanDuration / _expectedFrameDuration.inMilliseconds).ceil(),
+          slowFrameCount: 0,
+          frozenFrameCount: 0,
+          framesDelay: 0);
+    }
+
+    final spanStartMs = spanStartTimestamp.millisecondsSinceEpoch;
+    final spanEndMs = spanEndTimestamp.millisecondsSinceEpoch;
+    final expectedDurationMs = _expectedFrameDuration.inMilliseconds;
+    final frozenThresholdMs = _frozenFrameThreshold.inMilliseconds;
+
+    int slowFrameCount = 0;
+    int frozenFrameCount = 0;
+    int slowFramesDuration = 0;
+    int frozenFramesDuration = 0;
+    int framesDelay = 0;
+
+    for (final timing in delayedFrames) {
```

Comment:
Do we even need to iterate all delayed frames when we do have the `relevantFrames` from before for the span star/end? If not, we could search for the first relevant frame above and bail out early from the loop if we are only interested if there `are` relevant frames. Would be faster that way.

Reply:
> Do we even need to iterate all delayed frames when we do have the relevantFrames from before for the span star/end? If not

nope, good catch, this should be iterating the relevantFrames not all frames